src/back-end/lib/resources/affiliation.ts

```diff
@@ -103,19 +109,18 @@ const resource: Resource = {
           });
         }
       },
-      async respond(request): Promise<Response<JsonResponseBody<Affiliation | CreateValidationErrors>, Session>> {
-        const respond = (code: number, body: Affiliation | CreateValidationErrors) => basicResponse(code, request.session, makeJsonResponseBody(body));
-        switch (request.body.tag) {
-          case 'invalid':
-            if (request.body.value.permissions) {
-              return basicResponse(401, request.session, makeJsonResponseBody(request.body.value));
-            }
-            return basicResponse(400, request.session, makeJsonResponseBody(request.body.value));
-          case 'valid':
-            const affiliation = await createAffiliation(connection, request.body.value);
-            return respond(201, affiliation);
-        }
-      }
+      respond: wrapRespond<ValidatedCreateRequestBody, CreateValidationErrors, JsonResponseBody<Affiliation>, JsonResponseBody<CreateValidationErrors>, Session>({
```

Comment:
I had to specify the type parameters here to prevent the error I sent you earlier.  I still don't completely understand why, but I believe it was having trouble inferring a `ValidResB` from a `InvalidResB`.  Would be interested to know if you had any insight on why this was.  This only occurred on the create actions for `affiliation` and other resources.

Reply:
This is fine. TypeScript's type inference isn't good enough to figure out what the return type should be.